### PR TITLE
gh-87092: in compiler, move the detection of exception handlers before  the CFG optimization stage

### PR DESCRIPTION
**Pull request by iritkatriel**
* Issue: gh-87092

### CODE REVIEW

File: Python/compile.c [hunk 1]
```diff
@@ -7296,6 +7296,23 @@ label_exception_targets(basicblock *entryblock) {
     return -1;
 }
 
+
+static int
+mark_except_handlers(basicblock *entryblock) {
+    for (basicblock *b = entryblock; b != NULL; b = b->b_next) {
```
Comment (markshannon) [inline Python/compile.c:7303]: This gets evaluated in non debug mode. Maybe wrap in `#ifndef NDEBUG`?

File: Python/compile.c [hunk 2]
```diff
@@ -260,8 +260,6 @@ typedef struct basicblock_ {
     int b_ialloc;
     /* Number of predecssors that a block has. */
```
Comment (hauntsaninja) [inline Python/compile.c]: (unrelated typo)
```suggestion
    /* Number of predecessors that a block has. */
```

**Review by markshannon — CHANGES_REQUESTED** [review]
One minor issue, otherwise LGTM.

**Comment by iritkatriel** [comment]
I have made the requested changes; please review again.

**Pull request merged**